MASFoundation/Classes/_private_/services/model/MASModelService.m

```diff
@@ -2085,6 +2095,11 @@ - (void)loginAsRefreshTokenWithCompletion:(MASCompletionErrorBlock)completion
  */
 - (void)logout:(BOOL)force completion:(MASCompletionErrorBlock)completion
 {
+    //
+    // Reset auth providers
+    //
+    [self resetAuthProviders];
```

Comment:
we can avoid reseting authproviders in logout, so that we can avoid more calls to fetching authproviders.